packages/web-app-files/src/helpers/share/role.ts

```diff
@@ -193,6 +207,41 @@ export const linkRoleUploaderFolder = new LinkShareRole(
   $gettext('uploader'),
   [SharePermissions.create]
 )
+export const spaceViewer = new SpaceShareRole(
+  'spaceViewer',
+  false,
+  $gettext('Viewer'),
+  $gettext('viewer'),
+  [SharePermissions.read]
+)
+export const spaceEditor = new SpaceShareRole(
+  'spaceEditor',
+  false,
+  $gettext('Editor'),
+  $gettext('editor'),
+  [SharePermissions.read, SharePermissions.update, SharePermissions.create, SharePermissions.delete]
+)
+export const spaceManager = new SpaceShareRole(
+  'spaceManager',
+  false,
+  $gettext('Manager'),
+  $gettext('manager'),
+  [
+    SharePermissions.read,
+    SharePermissions.update,
+    SharePermissions.create,
+    SharePermissions.delete,
+    SharePermissions.share
+  ]
+)
+
+export abstract class SpacePeopleShareRoles {
+  static readonly all = [spaceViewer, spaceEditor, spaceManager]
+
```

Comment:
Please introduce a `list` method like for the other abstract collection classes. This way we can decide at any point in time to introduce whatever operations on the `all` array might be needed internally without changing references.

Reply:
I don't quite understand the `list` method yet. Should it just return all roles for now then?

Reply:
Yes, exactly. Just return `all`, but wrapped in a function. This gives us more flexibility for filtering etc without touching consuming code.

Reply:
Got it 👍